draft-dss-star.md

```diff
@@ -101,200 +121,640 @@ In this document we describe one such system, namely Distributed Secret Sharing
+key_prk = Extract(nil, key_seed)
+key = Expand(key_prk, "key", Nk)
+nonce = Expand(key_prk, "nonce", Nn)
+```
+
+The client then generates a secret share of `key_seed` using `share_coins` as randomness as follows:
+
+```
+rand_share = Share(REPORT_THRESHOLD, TBD, key_seed, share_coins, nil)
+```
+
+[[OPEN ISSUE: what should N be for the TSS scheme?]]
+
+The client then encrypts `msg` and `aux` using the KCAEAD key and nonce as follows:
+
+```
+report_data = msg || aux
+encrypted_report = Seal(key, nonce, nil, report_data)
+```
+
+Finally, the client constructs a report consisting of `encrypted_report`, `rand_share`,
+and `tag`, and sends this to the Anonymizing Server in the subsequent epoch, i.e., after
+the Randomness Server has rotated its OPRF key.
+
+[[OPEN ISSUE: how does the client know when this rotation happens?]]
+
+```
+struct {
+  opaque encrypted_report<1..2^16-1>;
+  opaque rand_share[Nshare];
+  opaque tag[16];
+} Report;
+```
+
+Specifically, Clients send a Report to the Aggregation Server using an HTTP POST message
+with content type "message/star-report". An example message is below.
+
+~~~
+:method = POST
+:scheme = https
+:authority = aggregator.example
+:path = /
+content-type = message/star-report
+content-length = <Length of body>
+
+<Bytes containing a Report>
+~~~
+
+This message is sent to the Aggregation Server through the Anonymizing Proxy. See {{proxy-options}}
+for different types of proxy options.
+
+## Aggregation Phase
+
+Aggregation is the final phase of STAR. It happens offline and does not require any
+communication between different STAR entities. It proceeds as follows. First, the
+Aggregation Server groups reports together based on their `tag` value. Let `report_set`
+denote a set of at least REPORT_THRESHOLD reports that have a matching `tag` value.
+
+Given this set, the Aggregation Server begins by running the secret share recovery algoritm
+as follows:
+
+```
+key_seed = Recover(report_set)
+```
+
+If this fails, the Aggregation Server chooses a new candidate report share set and
+reruns the aggregation process.
+
+[[OPEN ISSUE: how does the server choose new candidate sets when share recovery fails?]]
+
+Otherwise, the Aggregation Server derives the same KCAEAD key and nonce from `key_seed` to
+decrypt each of the report ciphertexts in `report_set`.
+
+```
+key_prk = Extract(nil, key_seed)
+key = Expand(key_prk, "key", Nk)
+nonce = Expand(key_prk, "nonce", Nn)
+```
+
+Each report ciphertext is decrypted as follows:
+
+```
+report_data = Seal(key, nonce, nil, ct)
+msg || aux = report_data
+```
+
+If this fails for any report, the Aggregation Server chooses a new candidate report share set and
+reruns the aggregation process. Otherwise, the Aggregation Server then outputs `msg` and each of
+the `aux` values for the corresponding reports.
+
+[[OPEN ISSUE: what happens if the msg is different for any two reports in the set? This should not happen if using a KCAEAD, but good to handle nevertheless]]
```

Comment:
Is this referring to if a client encrypts a different plaintext than the one used to create the share? If so, how would a KCAEAD help?

Reply:
A KCAEAD ensures that a given ciphertext can't decrypt to two different plaintexts under two different keys. So if the attacker can somehow force the server to use either of these keys, the server can't figure out which plaintext was correct. (This is somewhat tied to the secret sharing scheme.)